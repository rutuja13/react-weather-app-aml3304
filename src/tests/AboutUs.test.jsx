import { configure, shallow } from "enzyme";
import About from "../components/AboutUs";
import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({adapter: new Adapter()});

describe('About Us Component', () => {
    it('Snapshot Matching', () => {
       const wrapper = shallow(<About />);
       expect(wrapper.getElements()).toMatchSnapshot();
    });

    it('Prevent Users', () => {
       const wrapper = shallow(<About />);
       expect(wrapper.text().includes('shall not')).toBe(true);
    });

    it('Mentions College', () => {
        const wrapper = shallow(<About />);
        expect(wrapper.text().includes('Lambton College')).toBe(true);
     });

     it('Should not contain other college names', () => {
        const wrapper = shallow(<About />);
        expect(wrapper.text().includes('Seneca College')).toBe(false);
     });
 });