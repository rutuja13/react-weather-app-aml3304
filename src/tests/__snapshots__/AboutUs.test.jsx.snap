// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Us Component Snapshot Matching 1`] = `
Array [
  <div
    className="about"
  >
    <div
      class="container"
    >
      <div
        class="row align-items-center my-5"
      >
        <div
          class="col-lg-7"
        >
          <img
            alt=""
            class="img-fluid rounded mb-4 mb-lg-0"
            src="http://placehold.it/900x400"
          />
        </div>
        <div
          class="col-lg-5"
        >
          <h1
            class="font-weight-light"
          >
            About
          </h1>
          <p>
            We are a group of students studying with Lambton College and this application has been created as part of the project and shall not be used for any other purpose.
          </p>
        </div>
      </div>
    </div>
  </div>,
]
`;
